### PR TITLE
Add mingw support xvidcore

### PR DESCRIPTION
**Pull request by seanybaggins**
## Description of changes
Add mingw support for xvidcore

Part of a larger effort to add mingw support for qtmultimedia

Relevant for https://github.com/NixOS/nixpkgs/issues/274274

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/xvidcore/default.nix [hunk 1]
```diff
@@ -35,12 +35,17 @@ stdenv.mkDerivation rec {
     rm $out/lib/*.a
   '';
 
+  # Dependants of xvidcore don't know to look in bin for dependecies. Link them
+  # in lib so other depedants of xvidcore can find the dlls.
```
Comment (Artturin) [inline pkgs/development/libraries/xvidcore/default.nix:39]: Please give an example

Can the hook be improved https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/win-dll-link.sh

Comment (seanybaggins) [reply pkgs/development/libraries/xvidcore/default.nix:39]: ffmpeg is an example.

I think the hook can be improved. When creating my own windows application, I need to manually copy dlls to bin which is not ideal. See https://github.com/seanybaggins/qt-package-overalys/blob/bcd044bdba9be492e4694d096edb0bb4e9dbb027/nix/overlays/pkgs/qtapp-example/default.nix#L36-L43

**Comment by seanybaggins** [comment]
Why would this cause a massive rebuild. Things appear to be properly gated to me.

**Comment by seanybaggins** [comment]
@wegank

**Comment by Artturin** [comment]
> Why would this cause a massive rebuild. Things appear to be properly gated to me.

`optionalString` becomes `""` on false

File: pkgs/development/libraries/xvidcore/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/libraries/xvidcore/default.nix:39]: > I think the hook can be improved. When creating my own windows application, I need to manually copy dlls to bin which is not ideal. See [seanybaggins/qt-package-overalys@`bcd044b`/nix/overlays/pkgs/qtapp-example/default.nix#L36-L43](https://github.com/seanybaggins/qt-package-overalys/blob/bcd044bdba9be492e4694d096edb0bb4e9dbb027/nix/overlays/pkgs/qtapp-example/default.nix#L36-L43)
 
 The `/bin` copies/links should be done by the hook as long as you add the packages to `buildInputs`.

**Comment by seanybaggins** [comment]
Should this be rebased to staging-next or staging?

**Comment by Mindavi** [comment]
I think rebasing to -next would be ok but it should _target_ staginf

**Comment by seanybaggins** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/288877

**Pull request closed without merging**